runtime/runtime.m

```diff
@@ -2384,6 +2387,15 @@ -(void) xamarinSetFlags: (enum XamarinGCHandleFlags) flags;
 xamarin_vm_initialize ()
 {
 	char *pinvokeOverride = xamarin_strdup_printf ("%p", &xamarin_pinvoke_override);
+	char *icu_dat_file_path = NULL;
+
+	char path [1024];
+	if (!xamarin_locate_app_resource (xamarin_icu_dat_file_name, path, sizeof (path))) {
```

Comment:
That looks like `ios-arm` and `ios-arm64` each as a copy of the icu data file ?
I don't recall if those are identical (or not) between architectures... but if they are then it's a big opportunity for sharing it

Reply:
Yes, the icu data file is different between architectures (the arm one is bigger):

```
 ls -la [...]/xamarin-macios/tests/dotnet/packages/microsoft.netcore.app.runtime.mono.ios-arm*/6.0.0-preview.6.21276.13/runtimes/ios-arm*/native/icudt.dat
-rwxr--r--  1 rolf  wheel  1713152 May 18 21:54 [...]/xamarin-macios/tests/dotnet/packages/microsoft.netcore.app.runtime.mono.ios-arm/6.0.0-preview.6.21276.13/runtimes/ios-arm/native/icudt.dat
-rwxr--r--  1 rolf  wheel  1636880 May 18 21:54 [...]/xamarin-macios/tests/dotnet/packages/microsoft.netcore.app.runtime.mono.ios-arm64/6.0.0-preview.6.21276.13/runtimes/ios-arm64/native/icudt.dat
```

Reply:
The difference is unintentional. I don't think there's a tracking issue for it but I already mentioned it few times in some comments. The ICU data are generated from same input using the same tools. My theory is that there is some compression involved and the zlib library consumed by different builds is slightly different version with different defaults.

Reply:
Yeah that's unintended, they should be identical. Filed https://github.com/dotnet/runtime/issues/55637

Reply:
Thanks, I was about to file an issue and look into it.

Reply:
@filipnavara I wonder if ICU data for 64bit is a bit bigger due to some 64bit paddings etc for performance sake 🙂

Reply:
@EgorBo Unlikely. The arm64 ones on Andorid and iOS are different size (https://github.com/dotnet/designs/pull/225#discussion_r647159143). I don't think it's even memory mapped so the micro-optimizations would not make much difference.